client/src/components/Profile/EmploymentCard.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { List, Button, Form } from 'antd';
+import { ReadOutlined, FileAddOutlined } from '@ant-design/icons';
+import isEqual from 'lodash/isEqual';
+import CommonCardWithSettingsModal from './CommonCardWithSettingsModal';
+import { EmploymentRecordDto, UpdateProfileInfoDto } from 'api';
+import EmploymentHistoryFormItem from './EmploymentHistory/EmploymentHistoryFormItem';
+import EmploymentHistoryDisplayItem from './EmploymentHistory/EmploymentHistoryDisplayItem';
+import dayjs from 'dayjs';
+
+export type EmploymentRecordFormItem = Omit<EmploymentRecordDto, 'dateFrom' | 'dateTo'> & {
+  dateFrom: dayjs.Dayjs;
+  dateTo: dayjs.Dayjs;
+};
+
+type Props = {
+  data: EmploymentRecordDto[];
+  isEditingModeEnabled: boolean;
+  updateProfile: (data: UpdateProfileInfoDto) => Promise<boolean>;
+};
+
+const dateFormat = 'YYYY-MM-DD';
+
+const defaultEmploymentRecord: EmploymentRecordDto = {
+  title: '',
+  dateFrom: '',
+  dateTo: '',
+  toPresent: false,
+  companyName: '',
+  officeLocation: undefined,
+};
+
+const EmploymentCard = ({ isEditingModeEnabled, data, updateProfile }: Props) => {
+  const [form] = Form.useForm<{ employmentHistory: EmploymentRecordFormItem[] }>();
+  const values = Form.useWatch([], form);
+
+  const employmentRecordFormItemToDto = (employment: EmploymentRecordFormItem): EmploymentRecordDto => ({
+    ...employment,
+    dateFrom: employment.dateFrom.format(dateFormat),
+    dateTo: employment.toPresent ? '' : employment.dateTo.format(dateFormat),
+  });
+
+  const employmentRecordDtoToFormItem = (employment: EmploymentRecordDto): EmploymentRecordFormItem => ({
```

Comment:
do you need this functions to be defined inside component?

Reply:
Not really, thank you